analyzers/src/SonarAnalyzer.CSharp/Rules/ClassShouldNotBeEmpty.cs

```diff
@@ -30,12 +30,12 @@ public sealed class ClassShouldNotBeEmpty : ClassShouldNotBeEmptyBase<SyntaxKind
         && !typeDeclaration.Modifiers.Any(x => x.IsKind(SyntaxKind.PartialKeyword))
         && (node is ClassDeclarationSyntax || IsParameterlessRecord(node));
 
-    protected override BaseTypeDeclarationSyntax GetIfHasDeclaredBaseClass(SyntaxNode node) =>
+    protected override BaseTypeDeclarationSyntax GetIfHasDeclaredBaseClassOrInterface(SyntaxNode node) =>
         node is ClassDeclarationSyntax { BaseList: not null } declaration
             ? declaration
             : null;
 
-    protected override bool HasGenericBaseClassOrInterface(BaseTypeDeclarationSyntax declaration) =>
+    protected override bool HasInterfaceOrGenericBaseClass(BaseTypeDeclarationSyntax declaration) =>
```

Comment:
why did you rename this? 

it seems its implementation stayed the same for C# , but not for VB? why?

Reply:
I am confused because the check `classSymbol.Interfaces.Any()` is done inside the common class

and `HasInterfaceOrGenericBaseClass` doesn't seem to have changed behavior here - it looks at BaseList and GenericNameSyntax